hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestDatanodeReconfiguration.java

```diff
@@ -18,21 +18,94 @@
 
   @Test
   void reconfigurableProperties() {
-    assertProperties(getSubject(), ImmutableSet.of());
+    assertProperties(getSubject(), ImmutableSet.of(
+        HDDS_DATANODE_BLOCK_DELETING_LIMIT_PER_INTERVAL,
+        HDDS_DATANODE_BLOCK_DELETE_THREAD_MAX,
+        OZONE_BLOCK_DELETING_SERVICE_WORKERS));
+  }
+
+  @Test
+  public void blockDeletingLimitPerInterval() throws ReconfigurationException {
+    getFirstDatanode().getReconfigurationHandler().reconfigurePropertyImpl(
+        HDDS_DATANODE_BLOCK_DELETING_LIMIT_PER_INTERVAL, "1");
+
+    getFirstDatanode().getDatanodeStateMachine().getContainer()
+        .getBlockDeletingService().getBlockLimitPerInterval();
+
+    assertEquals(1, getFirstDatanode().getDatanodeStateMachine().getContainer()
+        .getBlockDeletingService().getBlockLimitPerInterval());
+  }
+
+  @Test
+  public void blockDeleteThreadMax() throws ReconfigurationException {
+    // Start the service and get the original pool size
+    ThreadPoolExecutor executor = ((DeleteBlocksCommandHandler)
+        getFirstDatanode().getDatanodeStateMachine().getCommandDispatcher()
+            .getDeleteBlocksCommandHandler()).getExecutor();
+    int originPoolSize = executor.getMaximumPoolSize();
+
+    // Attempt to increase the pool size by 1 and verify if it's successful
+    getFirstDatanode().getReconfigurationHandler().reconfigurePropertyImpl(
+        HDDS_DATANODE_BLOCK_DELETE_THREAD_MAX,
+        String.valueOf(originPoolSize + 1));
+    assertEquals(originPoolSize + 1, executor.getMaximumPoolSize());
+    assertEquals(originPoolSize + 1, executor.getCorePoolSize());
+
+    // Attempt to decrease the pool size by 1 and verify if it's successful
+    getFirstDatanode().getReconfigurationHandler().reconfigurePropertyImpl(
+        HDDS_DATANODE_BLOCK_DELETE_THREAD_MAX,
+        String.valueOf(originPoolSize - 1));
+    assertEquals(originPoolSize - 1, executor.getMaximumPoolSize());
+    assertEquals(originPoolSize - 1,  executor.getCorePoolSize());
+  }
+
+  @Test
+  public void blockDeletingServiceWorkers() throws ReconfigurationException {
+    ScheduledThreadPoolExecutor executor = (ScheduledThreadPoolExecutor)
+        getFirstDatanode().getDatanodeStateMachine().getContainer()
+            .getBlockDeletingService().getExecutorService();
+    int originPoolSize = executor.getCorePoolSize();
+
+    // Attempt to increase the pool size by 1 and verify if it's successful
+    getFirstDatanode().getReconfigurationHandler().reconfigurePropertyImpl(
+        OZONE_BLOCK_DELETING_SERVICE_WORKERS,
+        String.valueOf(originPoolSize + 1));
+    assertEquals(originPoolSize + 1, executor.getCorePoolSize());
+
+    // Attempt to decrease the pool size by 1 and verify if it's successful
+    getFirstDatanode().getReconfigurationHandler().reconfigurePropertyImpl(
+        OZONE_BLOCK_DELETING_SERVICE_WORKERS,
+        String.valueOf(originPoolSize - 1));
+    assertEquals(originPoolSize - 1, executor.getCorePoolSize());
+  }
```

Comment:
We can avoid duplication by converting to `@ParameterizedTest`: https://github.com/adoroszlai/hadoop-ozone/commit/5aeb768560b36bf182dcc96bc50a0c71625f279d

Also fixes findbugs error:

```
M D RV: Return value of org.apache.hadoop.ozone.container.keyvalue.statemachine.background.BlockDeletingService.getBlockLimitPerInterval() ignored, but method has no side effect  At TestDatanodeReconfiguration.java:[line 58]
```

https://github.com/xichen01/ozone/actions/runs/5577408467/jobs/10190274215#step:6:2367